src/plugins/expressions/server/mocks.ts

```diff
@@ -13,37 +13,28 @@ import { coreMock } from '../../../core/server/mocks';
 export type Setup = jest.Mocked<ExpressionsServerSetup>;
 export type Start = jest.Mocked<ExpressionsServerStart>;
 
-const createSetupContract = (): Setup => {
-  const setupContract: Setup = {
-    extract: jest.fn(),
-    fork: jest.fn(),
-    getFunction: jest.fn(),
-    getFunctions: jest.fn(),
-    getRenderer: jest.fn(),
-    getRenderers: jest.fn(),
-    getType: jest.fn(),
-    getTypes: jest.fn(),
-    inject: jest.fn(),
-    registerFunction: jest.fn(),
-    registerRenderer: jest.fn(),
-    registerType: jest.fn(),
-    run: jest.fn(),
-  };
-  return setupContract;
-};
-
-const createStartContract = (): Start => {
-  const startContract: Start = {
+const createSetupContract = (): Setup => ({
+  fork: jest.fn(),
+  getFork: jest.fn(),
+  getFunction: jest.fn(),
+  getFunctions: jest.fn(),
+  getRenderer: jest.fn(),
+  getRenderers: jest.fn(),
+  getType: jest.fn(),
+  getTypes: jest.fn(),
```

Comment:
we should most likely remove all the getters from setup contract as well as there is no way to be certain specific thing is going to be registered at the time you are requesting it (we can only guarantee that in start contract)

Reply:
I tried to remove that, but I found a few places where logic relies on those getters. I think it makes sense to keep them because it can be the case when you register functions conditionally.
- [x-pack/plugins/canvas/public/setup_expressions.ts](https://github.com/elastic/kibana/blob/master/x-pack/plugins/canvas/public/setup_expressions.ts).
- [x-pack/plugins/canvas/server/routes/functions/functions.ts](https://github.com/elastic/kibana/blob/master/x-pack/plugins/canvas/server/routes/functions/functions.ts).
- [x-pack/plugins/canvas/public/application.tsx](https://github.com/elastic/kibana/blob/master/x-pack/plugins/canvas/public/application.tsx).

Reply:
can we remove those that are not used, seems only `getFunction` is used at the moment ? 
and please open an issue for `getFunction` one. 
- the setup_expression.ts is not working reliably (possibly missing functions registered by other plugins)
- the functions.ts should be using start contract
- application.tsx should be using start contract.